dashboard/test/test_helper.rb

```diff
@@ -249,22 +249,62 @@ def assert_redirected_to_sign_in
   end
 
   def self.generate_admin_only_tests_for(action, params = {})
-    test "should get #{action}" do
-      sign_in create(:admin)
-      get action, params: params
-      assert_response :success
-    end
+    generate_user_tests_for action, user: :admin, params: params
+    generate_user_tests_for action, response: :forbidden, params: params
+    generate_redirect_to_sign_in_test_for action, params: params
+  end
+
+  # Generates a test case ensuring redirect to sign in for not signed in users
+  # @param action [String] the controller action to test
+  # @param method [Symbol, String] http method with which to perform the action (default :get)
+  # @param params [Hash, Proc] params to pass to the action. It can be a direct hash,
+  #   or a proc that generates a hash at runtime in the context of the test case.
+  def self.generate_redirect_to_sign_in_test_for(action, method: :get, params: {})
+    test "not signed in #{method} #{action} redirects to sign in" do
+      # params can be a hash, or a proc that will generate a hash at runtime
+      # (e.g. to create fixtures).
+      params = self.instance_exec(&params) if params.is_a? Proc
 
-    test "should not get #{action} if not signed in" do
       sign_out :user
-      get action, params: params
+      send method, action, params: params
       assert_redirected_to_sign_in
     end
+  end
 
-    test "should not get #{action} if not admin" do
-      sign_in create(:user)
-      get action, params: params
-      assert_response :forbidden
+  # Generates basic response validation test cases for logged-in users, one test case per specified user.
+  # @param action [String] the controller action to test
+  # @param method [Symbol, String] http method with which to perform the action (default :get)
+  # @param response [Symbol, String, Number] expected response (default :success)
+  # @param user [Symbol, String, Array<Symbol, String>, Proc] user to log in (default :user)
+  #   It can be a single or array of factory names to be passed to FactoryGirl.create,
+  #   or a proc that runs in the context of the test case and returns a user.
+  #   In the case of an array, one test case will be generated per user.
+  # @param params [Hash, Proc] params to pass to the action. It can be a direct hash,
+  #   or a proc that generates a hash at runtime in the context of the test case.
+  # @param name [String] optional name of the test case.
+  #   Otherwise it will be generated based on parameter values.
+  #   Note: It is recommended to supply a name when varying params or user procs,
+  #     as the default generated names may be ambiguous and may conflict.
+  def self.generate_user_tests_for(action, method: :get, response: :success,
```

Comment:
This seems to be doing too much, make the caller do some of the work. IMO, this should accept exactly one user (one test [suite] per thing [site] to be tested). And is it too much to ask that the caller instantiate the FactoryGirl or otherwise?